frontend/src/lib/components/TrainTracks.svelte

```diff
@@ -0,0 +1,108 @@
+<script lang="ts" context="module">
+    type Circle = {row:number, col:number};
+    type SVGDot = {x:number, y:number, color?:string};
+    type SVGStroke = {color?:string, d:string}
+</script>
+<script lang="ts">
+    // We need a list of dots and lines, i.e. which dots go to which parents
+    // Each dot has one or two parents, and needs a Bezier curve to the parent
+    export let paths: [number, number][] = [];  // Format is [0,1],[1,2],[2,3],[2,4]. Dot 0 connects to dot 1. Dot 1 connects to dot 2. Dot 2 connects to both both 3 and 4, but they are separate paths.
+    export let circles: Circle[] = [];  // Format is [{row:0,col:0},{row:1,col:0},{row:2,col:0},{row:2,col:1}]. Each dot has a row and column, which are auto-translated to x and y
```

Comment:
if I understand correctly the value of each path is a row index and a parent row index. Could we use an object with named properties instead of an array to make it a little more clear how paths work?

Reply:
Yes, a `{ from: number, to: number }` structure would make sense. Or `fromIdx` to make it more plain that they're indices.

And they're not indices into rows, they're indices into the circles list. Hence my reply (below) to your comment on line 94: sorting isn't necessary.